### DISCUSSION TITLE
Teleporting HSR while maintaining frame positions

### DISCUSSION THREAD

**Discussion by Kai-X-Org** [HumanSupportRobot/QandA#66] [Q&A]
Our project involves random initialization of robot position across many simulations, and we would like to be able to initialize the HSR position by using the SetModelPose service for gazebo. However, we realized that if we do that, the frames such as 'map' and 'odom' used by the HSR gets teleported as well and we get errors. Is there a way to suppress this behavior? Thanks a lot!

**Comment by masudam** [comment]
If you encounter errors, sharing the error logs will be helpful for troubleshooting. 
The appropriate response may vary depending on what you are trying to achieve. If you do not use the "map" and "odom" topics, you can simply ignore them. As for the "map" topic, it is possible to handle it by performing self-localization. Specifically, you can correct the map by sending the initial pose information of the HSR to the "/laser_2d_correct_pose" topic.

**Reply by Kai-X-Org** [reply]
Hi, 
I tried publishing to /laser_2d_correct_pose with the following code:

pose_w_cov_stamped = PoseWithCovarianceStamped()
pose_w_cov = pose_w_cov_stamped.pose
pose = pose_w_cov.pose
pose.position.x = -1.0
pose.position.y = -7.5
pose.position.z = 0.32
ori = quaternion_from_euler(0.0, 0.0, math.pi/2)
pose.orientation.x = ori[0]
pose.orientation.y = ori[1]
pose.orientation.z = ori[2]
pose.orientation.w = ori[3]
pose_w_cov_stamped.header.frame_id = 'map'
pose_w_cov_stamped.header.stamp=rospy.get_rostime()
print(pose_w_cov_stamped)
pub.publish(pose_w_cov_stamped)

But the message I see when doing rostopic echo /laser_2d_correct_pose is as shown in the screenshot, and nothing changes in the simulator. Thanks a lot!

<img width="1512" alt="Screenshot 2023-08-08 at 19 23 37" src="https://github.com/HumanSupportRobot/QandA/assets/70188506/e141b8d2-0cbd-4a0b-93e7-da367f186bb3">

**Reply by kosei1515** [reply]
Hi, could you please try the following code?
```
pose_w_cov_stamped = PoseWithCovarianceStamped()
pose_w_cov_stamped.pose.pose.position.x = -1.0
pose_w_cov_stamped.pose.pose.position.y = -7.5
pose_w_cov_stamped.pose.pose.position.z = 0.32
ori = quaternion_from_euler(0.0, 0.0, math.pi/2)
pose_w_cov_stamped.pose.pose.orientation.x = ori[0]
pose_w_cov_stamped.pose.pose.orientation.y = ori[1]
pose_w_cov_stamped.pose.pose.orientation.z = ori[2]
pose_w_cov_stamped.pose.pose.orientation.w = ori[3]
pose_w_cov_stamped.header.frame_id = 'map'
pose_w_cov_stamped.header.stamp=rospy.get_rostime()
print(pose_w_cov_stamped)
pub.publish(pose_w_cov_stamped)
```
Additionally, I would like you to share the result of `print` function shown on the console.
Thank you.

**Reply by Kai-X-Org** [reply]
Thank you so much! This seemed to have worked. If you still would like it, attached is the result of the print. Thanks a lot!
<img width="1113" alt="Screenshot 2023-08-13 at 00 14 33" src="https://github.com/HumanSupportRobot/QandA/assets/70188506/abc06e0a-12d2-4b83-a0fb-55bc6f2e875e">
